lib/auth/auth.go

```diff
@@ -4722,6 +4722,12 @@ func (a *Server) DeleteKubernetesCluster(ctx context.Context, name string) error
 
 // SubmitUsageEvent submits an external usage event.
 func (a *Server) SubmitUsageEvent(ctx context.Context, req *proto.SubmitUsageEventRequest) error {
+	event, err := usagereporter.ConvertUsageEventWithoutUser(req.GetEvent())
```

Comment:
Reviewers: I think this is necessary because the discovery server's identity doesn't have user metadata, but that could be because of how the tests are set up, so I'm not 100% sure. If that's not the case, please let me know!